File: src/my_py_pkg/my_py_pkg/add_two_ints_client.py
#!/usr/bin/env python3 
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

class Add_two_ints_client_node(Node):
    def __init__(self):
        super().__init__("Add_two_ints_client")
        self.client_=self.create_client(AddTwoInts,"add_two_ints")

    def call_add_int(self,a,b):
        while not self.client_.wait_for_service(1.0):
            self.get_logger().warn("waiting for the server...")
        request=AddTwoInts.Request()
        request.a=a
        request.b=b
        x=request
        future=self.client_.call_async(request)
        future.add_done_callback(lambda f: self.callback_call_add_two_ints(f,x))

    def callback_call_add_two_ints(self, future, x):
        response=future.result()
        self.get_logger().info(str(x.a) + " + " + str(x.b) + " = " + str(response.sum))
        rclpy.shutdown()


def main(args = None):
    rclpy.init(args=args)
    node =  Add_two_ints_client_node()
    node.call_add_int(3,8)
    rclpy.spin(node)
    if rclpy.ok():
        rclpy.shutdown()
if __name__=="__main__":
    main()

File: src/my_cpp_pkg/src/add_two_ints_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"
using namespace std::chrono_literals;
using namespace std::placeholders;

class add_two_ints_client_node : public rclcpp::Node
{
    public:
        add_two_ints_client_node() : Node("add_two_ints_client")
        {
            client_=this->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");


        }
        void call_f(int a, int b)
        {
           while (rclcpp::ok() && not client_ ->wait_for_service(1s))
                {
                    RCLCPP_WARN(this->get_logger(),"waiting for the server");

                     }

        auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
        request->a=a;
        request->b=b;
        
        client_->async_send_request(request, std::bind(&add_two_ints_client_node::callback_response,this,_1) );

            


        }
        
    private:
         rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedPtr client_;
         void callback_response(rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedFuture future)
         {
            auto response = future.get();
            RCLCPP_INFO(this->get_logger()," Sum = %d",(int)response->sum);
         }

};
int main(int argc, char **argv)
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<add_two_ints_client_node>();
    node->call_f(10,2);
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0 ;
}